### ISSUE TITLE
Add documentation section about proper type comparison (since v0.8)

### ISSUE THREAD

**Issue by Andrew15-5** [typst/typst#2177]
There is a [compatibility](https://typst.app/docs/reference/foundations/type/#compatibility) section on the `type` page, but it only shows how to compare `int` type. The reason for this issue is that I was unable to understand how to check for `content` type. Is it `type(var) == []`? `== type(content)`? <strike>Because comparing with the `content` doesn't work.</strike>

I'm not sure how many types there are in total, but I think at least the majority/often used ones should be covered.

**Update**:
I think I was making the wrong conditional expression assertion. But the `type(none)` is `none` (when rendered) while `type(none) == none` is `false`. Same with `auto`.

**Comment by laurmaedje** [comment]
Yeah, `none` and `auto` are a bit special because they are the only values of their type, their type isn't actual bound to any name. You'd rarely need it because `type(x) == type(none)` is the same as `x == none`. Some explanation for this would be useful.
